Dockerfile

```diff
@@ -24,10 +24,20 @@ RUN apt-get update && \
   && apt-get clean \
   && rmdir /var/cache/apt/archives/partial
 
+# Creating the user is required because jenkins runs he container
+# with the same user as the host (with '-u <uid>:<gid>')
+# but without the user existing 'git' fails with 'No user exists for uid ...'.
+ARG USER_ID=1000
+ARG GROUP_ID=1000
+RUN groupadd -g $GROUP_ID user && \
+    useradd -u $USER_ID -s /bin/sh -m -g user user
+
+COPY docker-entrypoint.sh /
 COPY src /app/src
 COPY flatpak-external-data-checker /app/
 COPY canonicalize-manifest /app/
 
 RUN python3 -m compileall /app/src
 
-ENTRYPOINT [ "/app/flatpak-external-data-checker" ]
+ENTRYPOINT [ "/docker-entrypoint.sh" ]
```

Comment:
Do you control docker cli args in Jenkins? If so, maybe it's better to pass `--entrypoint whatever` to `docker` instead of guessing what we should run in a shell script?

Reply:
Not sure tbh, the error comes from (which runs `cat` to test the container):
```
$ docker run -t -d -u 1004:1004 --privileged -w /home/jenkins/workspace/check-flatpak-external-apps-pipeline -v /home/jenkins/workspace/check-flatpak-external-apps-pipeline:/home/jenkins/workspace/check-flatpak-external-apps-pipeline:rw,z -v /home/jenkins/workspace/check-flatpak-external-apps-pipeline@tmp:/home/jenkins/workspace/check-flatpak-external-apps-pipeline@tmp:rw,z -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** flatpak-external-data-checker:latest cat
$ docker top 54624e2f05c7096d62a00b012b29e5ff3734393f613e885990d9c1c299799e9f -eo pid,comm
ERROR: The container started but didn't run the expected command. Please double check your ENTRYPOINT does execute the command passed as docker run argument, as required by official docker images (see https://github.com/docker-library/official-images#consistency for entrypoint consistency requirements).
```

I could try passing `--entrypoint` to `image.inside` on Jenkinsfile to see how if that would work, but wonder if it may still be useful (and best practice) to have an entrypoint allowing running arbitrary commands (bash for example for debugging shell)?

Reply:
>it may still be useful (and best practice) to have an entrypoint allowing running arbitrary commands

Since the first argument is likely to be a file with arbitrary name, such wrapper feels a bit fragile. But I'm not strictly against it.

Reply:
Pushed a new branch `fix-jenkins-integration-new` overriding the entrypoint via Jenkinsfile and will verify if it works and update here accordingly.

Reply:
Ok, it works, updated changes.